.github/workflows/main.yml

```diff
@@ -38,7 +38,7 @@ jobs:
      - name: Run presubmit tests
        run: |
             go env -w GOFLAGS=-mod=mod
-            make -j 5 all
+            make -j $(nproc) all
```

Comment:
Instead of `make -j all`, split them into individual jobs with the actions and they run parallel.

Reply:
IIUC, we need to maintain order in our build process. For example, `update-dependencies` step should always run before any other step while `tree-status` should be the last step. Similarly, some steps which generate files like proto and mocks should run before we build. Explicitly specifying this ordering in the Makefile is the best thing I could come up with. It's not ideal, but if folks have better ideas, please do let me know.

Reply:
> split them into individual jobs with the actions and they run parallel.

+1, example from Kubernetes RelEng: https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes-sigs/release-utils/release-utils-presubmits.yaml

There we usually prefer to run separate Prow jobs to split them up along the following boundaries:
- unit/integration tests
- e2e tests
- verifies (linting, `go mod tidy` diffs, boilerplate)